addons/youtube/__tests__/youtube.test.jsx

```diff
@@ -0,0 +1,59 @@
+const YouTube = require('../index')
+const Simulate = React.addons.TestUtils.Simulate
+const render = React.addons.TestUtils.renderIntoDocument
+
+describe('Addons - YouTube', function() {
+
+  it ('passes through slugs as a video_id content field', function(done) {
+    const slug = '0bRLtJHo0pI'
+    const test = function({ video_id }) {
+      video_id.should.equal(slug)
+      done()
+    }
+
+    const component = render(<YouTube onChange={ test } />)
+
+    component._onChange({ video_id: slug })
+  })
+
+  it ('can process undefined values', function(done) {
+    const test = function({ video_id }) {
+      video_id.should.equal('')
+      done()
+    }
+
+    const component = render(<YouTube onChange={ test } />)
+
+    component._onChange({ video_id: undefined })
+  })
+
+  describe('Extracting URLs', function() {
+    const patterns = {
+      shortcode: "http://youtu.be/dQw4w9WgXcQ",
+      embed: "http://www.youtube.com/embed/dQw4w9WgXcQ",
+      watch_query_string: "http://www.youtube.com/watch?v=dQw4w9WgXcQ",
+      query_string: "http://www.youtube.com/?v=dQw4w9WgXcQ",
+      v_url_parameter: "http://www.youtube.com/v/dQw4w9WgXcQ",
+      embed_url_parameter: "http://www.youtube.com/e/dQw4w9WgXcQ",
+      user_hash: "http://www.youtube.com/user/username#p/u/11/dQw4w9WgXcQ",
+      org_hash: "http://www.youtube.com/sandalsResorts#p/c/54B8C800269D7C1B/0/dQw4w9WgXcQ",
+      watch_extra_query_parameters: "http://www.youtube.com/watch?feature=player_embedded&v=dQw4w9WgXcQ",
+      extra_query_parameters: "http://www.youtube.com/?feature=player_embedded&v=dQw4w9WgXcQ"
+    }
+
+    for (let pattern in patterns) {
+      it (`extracts the video id from a YouTube ${ pattern.split("_").join(" ") }`, function(done) {
+        const url  = patterns[pattern]
+        const test = function({ video_id }) {
+          video_id.should.equal('dQw4w9WgXcQ')
+          done()
+        }
+
+        const component = render(<YouTube onChange={ test } />)
+
+        component._onChange({ video_id: url })
+      })
+    }
+  })
```

Comment:
Ah it just means that it's an async test. The test framework, Mocha, throws an error if the test times out after a specified time period (defaulting to 2 seconds) if done isn't called.

This works because `Function.prototype.length` returns the number of parameters a function expects. So like:

```javascript
function two_params (one, two) {}

console.log(two_params.length) // => 2
```
In Mocha's case, it sees that the function length is 1, so it sets up the test as asynchronous and passes in a callback as the first argument.